### PR TITLE
getRootFolder should not setup the FS for any user

### LINKED ISSUE #22467: Share file from ext storage from LDAP user to regular user, doesn't appear
### Steps to reproduce

1. Setup LDAP
1. Setup a SFTP external storage "/sftp"
1. Create a file "/sftp/test.txt"
1. Login as LDAP user
1. Share "/sftp/test.txt" with the local admin
1. Login as admin
1. Check if "test.txt" appears

### Expected behaviour
"test.txt" must be visible

### Actual behaviour
"test.txt not visible"

### Version
ownCloud 9.0 git master 7b0f83b616246c5274b551ab46b923b0989c464e

### Logs
```
{"reqId":"Or1fgaw0tn4oF3GlYh91","remoteAddr":"127.0.0.1","app":"files","message":" Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:44+00:00","method":"GET","url":"\/owncloud\/index.php\/apps\/files\/"}
{"reqId":"Or1fgaw0tn4oF3GlYh91","remoteAddr":"127.0.0.1","app":"core","message":"Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:44+00:00","method":"GET","url":"\/owncloud\/index.php\/apps\/files\/"}
{"reqId":"eqy29XkcrWkfpFWV4Ahk","remoteAddr":"127.0.0.1","app":"files","message":" Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:44+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/js\/oc.js?v=6954bd568d2f0d10c4f6a8d8da5d9c6d"}
{"reqId":"eqy29XkcrWkfpFWV4Ahk","remoteAddr":"127.0.0.1","app":"core","message":"Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:44+00:00","method":"GET","url":"\/owncloud\/index.php\/core\/js\/oc.js?v=6954bd568d2f0d10c4f6a8d8da5d9c6d"}
{"reqId":"ecpj+SE\/a\/X3IDmfdhUD","remoteAddr":"127.0.0.1","app":"files","message":" Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:45+00:00","method":"GET","url":"\/owncloud\/cron.php"}
{"reqId":"ecpj+SE\/a\/X3IDmfdhUD","remoteAddr":"127.0.0.1","app":"core","message":"Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:45+00:00","method":"GET","url":"\/owncloud\/cron.php"}
{"reqId":"Qv5hersvX7FQyEbXRnYZ","remoteAddr":"127.0.0.1","app":"files","message":" Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:45+00:00","method":"GET","url":"\/owncloud\/ocs\/v2.php\/apps\/notifications\/api\/v1\/notifications?format=json"}
{"reqId":"Qv5hersvX7FQyEbXRnYZ","remoteAddr":"127.0.0.1","app":"core","message":"Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:45+00:00","method":"GET","url":"\/owncloud\/ocs\/v2.php\/apps\/notifications\/api\/v1\/notifications?format=json"}
{"reqId":"8vDkxFfMVUZHOQd1wqjq","remoteAddr":"127.0.0.1","app":"files","message":" Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:45+00:00","method":"GET","url":"\/owncloud\/index.php\/avatar\/root\/28"}
{"reqId":"8vDkxFfMVUZHOQd1wqjq","remoteAddr":"127.0.0.1","app":"core","message":"Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:45+00:00","method":"GET","url":"\/owncloud\/index.php\/avatar\/root\/28"}
{"reqId":"cBTW4V2b+b9ZpX2jOqlG","remoteAddr":"127.0.0.1","app":"files","message":" Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:46+00:00","method":"PROPFIND","url":"\/owncloud\/remote.php\/webdav\/"}
{"reqId":"cBTW4V2b+b9ZpX2jOqlG","remoteAddr":"127.0.0.1","app":"core","message":"Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:46+00:00","method":"PROPFIND","url":"\/owncloud\/remote.php\/webdav\/"}
{"reqId":"JfKgvENdzSuOBirAvoHw","remoteAddr":"127.0.0.1","app":"files","message":" Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:46+00:00","method":"GET","url":"\/owncloud\/index.php\/avatar\/root\/28"}
{"reqId":"JfKgvENdzSuOBirAvoHw","remoteAddr":"127.0.0.1","app":"core","message":"Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:46+00:00","method":"GET","url":"\/owncloud\/index.php\/avatar\/root\/28"}
{"reqId":"7dtbgfZg0FA5c91TMH6Q","remoteAddr":"127.0.0.1","app":"files","message":" Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:46+00:00","method":"GET","url":"\/owncloud\/ocs\/v1.php\/apps\/files_external\/api\/v1\/mounts?format=json"}
{"reqId":"7dtbgfZg0FA5c91TMH6Q","remoteAddr":"127.0.0.1","app":"core","message":"Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:46+00:00","method":"GET","url":"\/owncloud\/ocs\/v1.php\/apps\/files_external\/api\/v1\/mounts?format=json"}
{"reqId":"W+rtpgJI0CtXM+YUukps","remoteAddr":"127.0.0.1","app":"files","message":" Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:46+00:00","method":"GET","url":"\/owncloud\/ocs\/v2.php\/apps\/files_sharing\/api\/v1\/shares?subfiles=true&path=%2F&format=json"}
{"reqId":"W+rtpgJI0CtXM+YUukps","remoteAddr":"127.0.0.1","app":"core","message":"Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:46+00:00","method":"GET","url":"\/owncloud\/ocs\/v2.php\/apps\/files_sharing\/api\/v1\/shares?subfiles=true&path=%2F&format=json"}
{"reqId":"+VUUu9SuANuu533IK11O","remoteAddr":"127.0.0.1","app":"files","message":" Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:46+00:00","method":"GET","url":"\/owncloud\/index.php\/apps\/files\/ajax\/getstoragestats.php?dir=%2F"}
{"reqId":"+VUUu9SuANuu533IK11O","remoteAddr":"127.0.0.1","app":"core","message":"Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:46+00:00","method":"GET","url":"\/owncloud\/index.php\/apps\/files\/ajax\/getstoragestats.php?dir=%2F"}
{"reqId":"4nWmWjwIk3dbtp7sLEJJ","remoteAddr":"127.0.0.1","app":"files","message":" Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:46+00:00","method":"GET","url":"\/owncloud\/index.php\/apps\/files_external\/userglobalstorages\/1"}
{"reqId":"4nWmWjwIk3dbtp7sLEJJ","remoteAddr":"127.0.0.1","app":"core","message":"Backends provided no user object for 0868ad0a-69b1-1035-907c-63e3f77fca33","level":3,"time":"2016-02-17T12:42:46+00:00","method":"GET","url":"\/owncloud\/index.php\/apps\/files_external\/userglobalstorages\/1"}
```

Note: also happens if you limit the applicables to the LDAP user.

That LDAP id is from the sharing LDAP user:
```
*************************** 1. row ***************************
       ldap_dn: uid=zombie0,ou=zombies,dc=owncloud,dc=com
 owncloud_name: 0868ad0a-69b1-1035-907c-63e3f77fca33
directory_uuid: 0868ad0a-69b1-1035-907c-63e3f77fca33
```

@davitol @icewind1991 @rullzer @blizzz 

I suspect an issue with initializing received share mounts.
Could also be an issue where the wrong user id format is used when passed to the user manager ?

@davitol please confirm whether this is a regression from v8.2.2

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #22467

This can go wrong when an app (take the ldap app) DIs something that
needs the rootFolder. This break if we use cookie auth since then we
know the user at that point and thus try to setup the fs for that user.

However if there are then incomming shares from an ldap user they will
fails since the user manager can't find them yet.

Now getRootFolder does not setup the fs for any user.

@icewind1991 I plese have a look at this.

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
Scrutinizer — 2 new issues 

@rullzer - https://scrutinizer-ci.com/g/owncloud/core/inspections/7d2bf7d3-fd77-41f2-b92c-c00808f452f0

**Comment by icewind1991** [comment]
I think this should be fine :+1:

**Comment by rullzer** [comment]
Please review: @PVince81 @nickvergessen @DeepDiver1975

File: lib/private/files/node/root.php [hunk 1]
```diff
@@ -71,7 +71,7 @@ class Root extends Folder implements IRootFolder {
 	/**
 	 * @param \OC\Files\Mount\Manager $manager
 	 * @param \OC\Files\View $view
-	 * @param \OC\User\User $user
+	 * @param \OC\User\User | null $user
```
Comment (nickvergessen) [inline lib/private/files/node/root.php]: not sure, does spacing kill type documentation?
http://manual.phpdoc.org/HTMLSmartyConverter/HandS/phpDocumentor/tutorial_tags.param.pkg.html

This only says that `a|b` works

Comment (rullzer) [inline lib/private/files/node/root.php]: mmm ok.. fixed.

**Comment by PVince81** [comment]
Tested, works :+1:

**Comment by PVince81** [comment]
Note: this assumes that API consumers are using `getUserFolder` instead of `getRootFolder`, because from `getRootFolder` they would not be able to reach the user's homes.

**Comment by PVince81** [comment]
Potentially dangerous, see https://github.com/owncloud/core/pull/22410/files#r53318501

@icewind1991 please re-confirm

**Comment by rullzer** [comment]
> Potentially dangerous, see https://github.com/owncloud/core/pull/22410/files#r53318501

I don't think it is. If you want to access a user folder you should init the mount points.. This logic does not change that.

But indeed @icewind1991 please double check.

**Comment by icewind1991** [comment]
#22410 manually sets up the fs so it's unaffected

**Comment by PVince81** [comment]
Right, makes sense. Thanks

**Pull request merged**